### PR TITLE
Create Libraries section and add v4-utils-js typedocs 

### PR DESCRIPTION
**Pull request by kamescg**

### CODE REVIEW

**Comment by asselstine** [comment]
Remember to follow the existing conventions!

- The title `v4-utils-js` should be formatted like the rest, not snake case
- The package should be under "technical reference", not alone by itself in a Libraries menu item.

**Comment by kamescg** [comment]
> Remember to follow the existing conventions!
> 
> * The title `v4-utils-js` should be formatted like the rest, not snake case
> * The package should be under "technical reference", not alone by itself in a Libraries menu item.

Yes for sure @asselstine  - this PR should really be a draft right now. I still need to format and wanted to sync with @ddesrosier regarding standardizing the typedoc formatting when we migrate generated docs.

Also I don't think the JS libraries should be under technical reference - that seems redundantly nested. It's all technical references 😉  Also from a Developer Experience perspective I personally would find "Smart Contracts" and "Libraries" top-level namespaces more understandable.

File: docs/libraries/v4-utils-js/index.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+The `@pooltogether/v4-utils-js` [node module package](https://www.npmjs.com/package/@pooltogether/v4-utils-js) provides calculations, computations and general utility functions for the PoolTogether V4 protocol.
+
+As normally the case, the utility library was designed to be modular. Enabling developers to more easily use low-level primitives to create the higher-level computations/operations. High-order operations like `winningPicks()` which takes an `address`, plus `Draw` and `PrizeDistribution` EVM structs fetched via `v4-js-client`: returning encoded transactions **ready for any EVM network**.
+
+Helping with both **low-level calculations** and **higher-order computations**, plus **essential utilities** like encoding, filtering and sorting.
+
+**Join the PoolTogether Discord, ask questions and get help from the community.**
+
+[![Discord](https://badgen.net/badge/icon/discord?icon=discord&label)](https://discord.gg/JFBPMxv5tr)
```
Comment (dylandesrosier) [inline docs/libraries/v4-utils-js/index.md:16]: We have discord invites turned off so you need to send the users to pooltogether.com/discord

File: docs/libraries/v4-utils-js/index.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+# Utility Library
+![Tests](https://github.com/pooltogether/v4-utils-js/actions/workflows/main.yml/badge.svg)
+[![Coverage Status](https://coveralls.io/repos/github/pooltogether/v4-utils-js/badge.svg?branch=main)](https://coveralls.io/github/pooltogether/v4-utils-js?branch=main)
+![ts](https://badgen.net/badge/-/TypeScript?icon=typescript&label&labelColor=blue&color=555555)
+[![GPLv3 license](https://img.shields.io/badge/License-GPLv3-blue.svg)](http://perso.crans.org/besson/LICENSE.html)
+![npm](https://img.shields.io/npm/v/@pooltogether/v4-utils-js)
+
+The `@pooltogether/v4-utils-js` [node module package](https://www.npmjs.com/package/@pooltogether/v4-utils-js) provides calculations, computations and general utility functions for the PoolTogether V4 protocol.
+
+As normally the case, the utility library was designed to be modular. Enabling developers to more easily use low-level primitives to create the higher-level computations/operations. High-order operations like `winningPicks()` which takes an `address`, plus `Draw` and `PrizeDistribution` EVM structs fetched via `v4-js-client`: returning encoded transactions **ready for any EVM network**.
```
Comment (dylandesrosier) [inline docs/libraries/v4-utils-js/index.md:10]: > As normally the case...

> Helping with both...

I don't know what these 2 paragraphs add beyond the first one.

File: docs/libraries/v4-utils-js/index.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+Helping with both **low-level calculations** and **higher-order computations**, plus **essential utilities** like encoding, filtering and sorting.
+
+**Join the PoolTogether Discord, ask questions and get help from the community.**
+
+[![Discord](https://badgen.net/badge/icon/discord?icon=discord&label)](https://discord.gg/JFBPMxv5tr)
+
```
Comment (dylandesrosier) [inline docs/libraries/v4-utils-js/index.md:17]: A "Table of contents" header would be nice here

File: docs/libraries/v4-utils-js/index.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+**Calculate**
```
Comment (dylandesrosier) [inline docs/libraries/v4-utils-js/index.md:18]: A sentence explaining the difference between "calculate" and "compute" within the context of this library would help clarify these sections/functions.

I think links to `v4-utils-js/calculate`,`v4-utils-js/compute` and `v4-utils-js/utilities` and a sentence explaining each is a better entry to the docs vs every function on the main page. Maintaining that will suck, and it's kind of a scary overload of functions.

File: docs/libraries/v4-utils-js/index.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+```ts
+import { computeWinningPicks, encodeWinningPicks } from '@pooltogether/v4-utils-js';
+const computedPicks = computeWinningPicks(wallet.address, [draw], [prizeDistribution]);
+const transaction = encodeWinningPicks(wallet.address, computedWinningPicks);
+wallet.send(transaction.encodedWinningPickIndices)
+```
+
+## 🧮 &nbsp;Examples
+
+The utility library simulates smart contract rules/operations and also encapsulates higher-level abstractions common to PoolTogether V4 required transactions.
```
Comment (dylandesrosier) [inline docs/libraries/v4-utils-js/index.md:94]: > The utility library simulates smart contract rules/operations and also encapsulates higher-level abstractions common to PoolTogether V4 required transactions.

I think this sentence fits perfectly in the opening paragraph for this library.

File: docs/libraries/v4-utils-js/compute.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
+#  compute
```
Comment (dylandesrosier) [inline docs/libraries/v4-utils-js/compute.md:1]: Idk if we can get this to generate properly, but having this file name (and the other pages) capitalized makes the sidebar look way nicer imo

**Pull request merged**